internal/processrunner/readychecker.go

```diff
@@ -0,0 +1,71 @@
+package processrunner
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// ReadyChecker is a struct used within the ProcessRunner to check if the
+// process being run has completed all its startup steps.
+type ReadyChecker struct {
+	callBackName  string
+	callback      func() bool
+	retryCount    int
+	retryInterval time.Duration
+	haltIfFailed  bool
+}
+
+// ReadyCheckerOpts is a struct used to pass options into NewReadyChecker.
+type ReadyCheckerOpts struct {
+	// CallBackName is the name of the callback function and is used for logging purposes.
+	CallBackName string
+	// Callback is the anonymous function that will be called to check if all
+	// startup requirements for the process have been completed. The function
+	// should return true if all startup checks are complete, and false if any
+	// startup checks have not completed.
+	Callback      func() bool
+	RetryCount    int
+	RetryInterval time.Duration
+	// HaltIfFailed is a flag that determines if the process should halt the app or
```

Comment:
Docs nit but I think these helpful comments describing the fields should go on the struct definition where the fields are defined. I think that's where a newcomer would be likely to look first.